processing/src/main/java/io/druid/query/groupby/epinephelinae/SpillingGrouper.java

```diff
@@ -99,6 +103,25 @@ public SpillingGrouper(
           limitSpec.getLimit(),
           sortHasNonGroupingFields
       );
+      // if configured buffer size is too small to support limit push down, don't apply that optmization
+      if (!limitGrouper.validateBufferCapacity(mergeBufferSize)) {
+        if (sortHasNonGroupingFields) {
+          log.info("Ignoring forceLimitPushDown, insufficient buffer capacity.");
+        }
+        this.grouper = new BufferHashGrouper<>(
+            bufferSupplier,
+            keySerde,
+            columnSelectorFactory,
+            aggregatorFactories,
+            bufferGrouperMaxSize,
+            bufferGrouperMaxLoadFactor,
+            bufferGrouperInitialBuckets,
+            // when there are non-grouping fields, we need to sort by a dimension-only ordering for merging to work
```

Comment:
This comment seems strange to me. Don't we _always_ need to sort by a dimension-only ordering for merging, regardless of whether or not there are non-grouping fields?

Reply:
Clarified the comment